src/Aop/Framework/ClassFieldAccess.php

```diff
@@ -117,14 +117,20 @@ public function &getValueToSet()
      */
     public function ensureScopeRule(int $stackLevel = 2): void
     {
-        $property = $this->reflectionProperty;
-
-        if ($property->isProtected()) {
+        $property    = $this->reflectionProperty;
+        $isProtected = $property->isProtected();
+        $isPrivate   = $property->isPrivate();
+        if ($isProtected || $isPrivate) {
             $backTrace     = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, $stackLevel+1);
             $accessor      = $backTrace[$stackLevel] ?? [];
             $propertyClass = $property->class;
             if (isset($accessor['class'])) {
-                if ($accessor['class'] === $propertyClass || is_subclass_of($accessor['class'], $propertyClass)) {
+                // For private and protected properties its ok to access from the same class
+                if ($accessor['class'] === $propertyClass) {
+                    return;
+                }
+                // For protected properties its ok to access from any subclass
+                if ($isProtected && is_subclass_of($accessor['class'], $propertyClass)) {
                     return;
                 }
```

Comment:
The exception message in line 137 below is probably no longer accurate: `Cannot access protected property...`.

Reply:
Thanks, somehow have forgotten about this line )